dali/kernels/imgproc/geom/remap_npp.h

```diff
@@ -0,0 +1,162 @@
+  using SupportedInputTypes = std::tuple<uint8_t, int16_t, uint16_t, float>;
+  static_assert(has_type_v<T, SupportedInputTypes>, "Unsupported input type.");
+
+
+  void Run(KernelContext &context,
+           TensorListView<Backend, T> output,
+           TensorListView<Backend, const T> input,
+           TensorListView<Backend, const MapType, 2> mapsx,
+           TensorListView<Backend, const MapType, 2> mapsy,
+           span<Box<2, int64_t>> output_rois = {},
+           span<Box<2, int64_t>> input_rois = {},
+           span<DALIInterpType> interpolations = {},
+           span<Border> borders = {}) override {
+    CUDA_CALL(nppSetStream(context.gpu.stream));
+    NppStreamContext npp_ctx;
+    CUDA_CALL(nppGetStreamContext(&npp_ctx));
+    for (int sample_id = 0; sample_id < input.num_samples(); sample_id++) {
+      const auto &inp = input[sample_id];
+      const auto &out = output[sample_id];
+      const auto &mx = mapsx[sample_id];
+      const auto &my = mapsy[sample_id];
+      auto oroi = !output_rois.empty() ? output_rois[sample_id] : detail::default_roi(inp.shape);
+      auto iroi = !input_rois.empty() ? input_rois[sample_id] : detail::default_roi(inp.shape);
+      auto interp = !interpolations.empty() ? interpolations[sample_id] : DALI_INTERP_LINEAR;
+
+      invoke_remap_kernel(inp, out, mx, my, iroi, oroi, to_npp(interp), npp_ctx);
+    }
+  }
+
+
+  void Run(KernelContext &context,
+           TensorListView<Backend, T> output,
+           TensorListView<Backend, const T> input,
+           TensorView<Backend, const MapType, 2> mapx,
+           TensorView<Backend, const MapType, 2> mapy,
+           Box<2, int64_t> output_roi = {},
+           Box<2, int64_t> input_roi = {},
+           DALIInterpType interpolation = DALI_INTERP_LINEAR,
+           Border border = {}) override {
+    CUDA_CALL(nppSetStream(context.gpu.stream));
+    NppStreamContext npp_ctx;
+    CUDA_CALL(nppGetStreamContext(&npp_ctx));
+    for (int sample_id = 0; sample_id < input.num_samples(); sample_id++) {
+      const auto &inp = input[sample_id];
+      const auto &out = output[sample_id];
+      invoke_remap_kernel(inp, out, mapx, mapy,
+                          !input_roi.empty() ? input_roi : detail::default_roi(inp.shape),
+                          !output_roi.empty() ? output_roi : detail::default_roi(inp.shape),
+                          to_npp(interpolation), npp_ctx);
+    }
+  }
+
+
+ private:
+  void invoke_remap_kernel(TensorView<Backend, const T> input,
+                           TensorView<Backend, T> output,
+                           TensorView<Backend, const MapType, 2> mapx,
+                           TensorView<Backend, const MapType, 2> mapy,
+                           Box<2, int64_t> input_roi,
+                           Box<2, int64_t> output_roi,
+                           int interpolation, NppStreamContext ctx) {
+    assert(input.shape == output.shape);
```

Comment:
Is it really a necessary constrain?

Reply:
I believe that yes. No matter if the ROI is specified or not, the output size should always match the input size. Unless you believe otherwise?

Reply:
It's not necessary and it's not the right check.
The output must have the same spatial dimensions as the _mapping_, not the input.
Ultimately, output elements are defined as:
`out[pos] = sample_at(input, ivec2(mapping_x[pos], mapping_y[pos]))`

Reply:
Corrected the checks